### ISSUE TITLE
OpenCV 2.4.13.2 fails to build

### ISSUE THREAD

**Issue by zbeekman** [Homebrew/homebrew-science#6151]
### Indicate troubleshooting steps taken

- [X] Ran `brew update` and retried your prior step?
- [X] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [X] Checked that a similar issue hasn't already been filed?
- [X] Looked up the software bug tracker if reporting a build error?
- [X] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports / build errors:

CMake syntax error: `brew upgrade --verbose --with-ffmpeg opencv` 

```
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2")
-- Looking for ccache - found (/usr/local/bin/ccache)
CMake Error at cmake/OpenCVCompilerOptions.cmake:21 (else):
  A duplicate ELSE command was found inside an IF block.
Call Stack (most recent call first):
  CMakeLists.txt:432 (include)


-- Configuring incomplete, errors occurred!
```

#### Brief summary of issue:

CMake is broken for OpenCV 2.4.13.2 (at least when ccache is installed), and needs to be patched or updated. Here is the relevant fix: https://github.com/opencv/opencv/commit/c2e281f8cf1600756d396ea4e969222931d704dc

#### The complete command that you issued:

`brew upgrade --verbose --with-ffmpeg opencv` 

#### Gist with output of `brew doctor`:

```
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/python2.7/greenlet/greenlet.h
  /usr/local/include/python2.7/zope.proxy/proxy.h

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  open-mpi
```
#### Link output by `brew gist-logs <formula>`:

https://gist.github.com/c3074f34abcfcf51ca9e5098b440a4df

#### Output of `brew config`:

```
$ brew config
HOMEBREW_VERSION: 1.3.1-62-g3b92f69
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 3b92f69869f0bc88c2f99854392535d4d522902c
Last commit: 41 minutes ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 6280a59d0a5472b144318dabedf07ff660f323fc
Core tap last commit: 25 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit sandybridge
Homebrew Ruby: 2.0.0-p481
GCC-4.2: build 5666
Clang: 7.0 build 700
Git: 2.14.1 => /usr/local/bin/git
Perl: /usr/local/bin/perl => /usr/local/Cellar/perl/5.26.0/bin/perl
Python: /usr/local/opt/python/libexec/bin/python => /usr/local/Cellar/python/2.7.13_1/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.1_1/bin/ruby
Java: 1.8.0_144, 1.7.0_80, 1.6.0_65-b14-468
macOS: 10.10.5-x86_64
Xcode: 7.2.1
CLT: 7.2.0.0.1.1447826929
X11: 2.7.11 => /opt/X11
```

**Comment by ilovezfs** [comment]
@zbeekman do you mind opening a PR against their 2.4 branch? It looks like the fix hasn't been backported yet. Thanks!

**Comment by iMichka** [comment]
Patch has been merged to the 2.4 branch. Either there will be a new release soon and the patch comes with it; or you can open a pull request to get `opencv@2` patched in homebrew-core (as opencv was migrated there): https://github.com/Homebrew/homebrew-core

Closing the current issue as the formula is gone from science :)
